### PR TITLE
Add a 5-second timeout at the server

### PR DESCRIPTION
**Pull request by diegobernardes**

### CODE REVIEW

File: internal/transport/server.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func (s *Server) Stop(ctx context.Context) error {
 func (s *Server) initMiddleware() {
 	m := middleware{log: s.Logger, writer: s.writer, traceExtractor: s.TraceExtractor}
 	s.router.Use(m.recoverer)
+	s.router.Use(m.timeout(5 * time.Second))
```
Comment (kevinmartingrant) [inline internal/transport/server.go:79]: why is this needed?

**Review by kevinmartingrant — APPROVED** [review]
LGTM once the question is answered.

File: internal/transport/server.go (hunk 1 above)
Comment (diegobernardes) [reply internal/transport/server.go:79]: To limit the max time a request can be at the server. I'm seeing huge latency spikes at the metrics, this will make sure those spikes will reach 5 seconds at the maximum and help the user because after 5 seconds this is likely to fail anyway.
